docs/2.9. Usage - Fully Automated Testing of NMOS Controllers.md

```diff
@@ -0,0 +1,174 @@
+    
+    "description": "String, a brief description of the test"
+    
+    "question": "String, human readable question text containing instructions to carry out the test"
+    
+    "answers": "Array containing answer objects to be used to select the correct answer_id to be returned.",
+        "type": "array",
+        "example":
+            "answer_id": "answer_2",
+            "label": "s3/wright",
+            "description": "Mock sender 3",
+            "id": "71ce775c-cdc5-479a-bd29-87c081905ecf",
+            "answer_str": "s3/wright (Mock sender 3, 71ce775c-cdc5-479a-bd29-87c081905ecf)"
+        "items": 
+            "required": "answer_id", "label", "description", "id", "answer_str"
+            "properties": 
+                "answer_id": "String, unique to each answer within a question. This is the value to be returned for validation."
+                "label": "Resource label as might be displayed on the NCuT"
+                "description": "Resource description as might be displayed on the NCuT"
+                "id": "Resource ID as might be displayed on the NCuT"
+                "answer_str": "String combining all the resource details for display on the testing facade"
+    
+    "time_sent": "Float, time the question was sent"
+    
+    "timeout": "Float, time in seconds the test suite will wait to receive an answer to a particular question",
+        "default": 600
+    
+    "url_for_response": "URL of the test suite API endpoint to send the POST request with answers",
+
+    "answer_response": "Null when received. Should be filled with answer_ids or 'Next' according to question type when POSTed back to url_for_response",
+        "default": "",
+        "anyOf":   
+            "null": "Null entry when sent from test suite",
+            "answer_1": "A single answer string, for 'radio' type questions",
+            ["answer_1", "answer_5"]: "Multiple answer strings, for 'checkbox' type questions",
+            "Next": "Specific string 'Next', for 'action' type questions"
+                
+    "time_answered": "Null when received. Should be filled with float string of time answered when returned",
+        "default": ""
+    
+    "metadata": "Details of Sender and Receiver where specified in the question text for IS05 tests",
+        "properties": 
+            "sender": "Details of sender chosen to be part of IS05 connection management test",
+                "required": "id", "label", "description"
+                "properties": 
+                    "id": "ID of sender as might be shown on NCut"
+                    "label": "Label of sender as might be shown on NCuT"
+                    "description": "Description of sender as might be shown on NCuT"
+            "receiver": "Details of receiver chosen to be part of IS05 connection management test. Same properties as sender"
+```
+Example radio question POST request from test suite to `/x-nmos/testing-facade`:
```

Comment:
Generally, I'd prefer to see a single blank line between all headings, paragraphs, code blocks, etc. It makes it a little easier to scan the Markdown and some Markdown tools do better.

Reply:
OK noted